macOS linking git fails

```
Error: The `brew link` step did not complete successfully
[721](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:721)
The formula built, but is not symlinked into /usr/local
[722](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:722)
Could not symlink etc/bash_completion.d/git-completion.bash
[723](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:723)
Target /usr/local/etc/bash_completion.d/git-completion.bash
[724](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:724)
is a symlink belonging to git@2.35.1. You can unlink it:
[725](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:725)
  brew unlink git@2.35.1
[726](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:726)

[727](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:727)
To force the link and overwrite all conflicting files:
[728](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:728)
  brew link --overwrite git
[729](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:729)

[730](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:730)
To list all files that would be deleted:
[731](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:731)
  brew link --overwrite --dry-run git
[732](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:732)

[733](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:733)
Possible conflicting files are:
[734](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:734)
/usr/local/etc/bash_completion.d/git-completion.bash -> /usr/local/Cellar/git@2.35.1/2.35.1/etc/bash_completion.d/git-completion.bash
[735](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:735)
/usr/local/etc/bash_completion.d/git-prompt.sh -> /usr/local/Cellar/git@2.35.1/2.35.1/etc/bash_completion.d/git-prompt.sh
[736](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:736)
/usr/local/bin/git -> /usr/local/Cellar/git@2.35.1/2.35.1/bin/git
[737](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:737)
/usr/local/bin/git-cvsserver -> /usr/local/Cellar/git@2.35.1/2.35.1/bin/git-cvsserver
[738](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:738)
/usr/local/bin/git-receive-pack -> /usr/local/Cellar/git@2.35.1/2.35.1/bin/git-receive-pack
[739](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:739)
/usr/local/bin/git-shell -> /usr/local/Cellar/git@2.35.1/2.35.1/bin/git-shell
[740](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:740)
/usr/local/bin/git-upload-archive -> /usr/local/Cellar/git@2.35.1/2.35.1/bin/git-upload-archive
[741](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:741)
/usr/local/bin/git-upload-pack -> /usr/local/Cellar/git@2.35.1/2.35.1/bin/git-upload-pack
[742](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:742)
Error: Could not symlink share/doc/git-doc/MyFirstContribution.html
[743](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:743)
Target /usr/local/share/doc/git-doc/MyFirstContribution.html
[744](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:744)
is a symlink belonging to git@2.35.1. You can unlink it:
[745](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:745)
  brew unlink git@2.35.1
[746](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:746)

[747](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:747)
To force the link and overwrite all conflicting files:
[748](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:748)
  brew link --overwrite git@2.35.1
[749](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:749)

[750](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:750)
To list all files that would be deleted:
[751](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:751)
  brew link --overwrite --dry-run git@2.35.1
[752](https://github.com/kaihowl/dotfiles/runs/6148553790?check_suite_focus=true#step:3:752)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -39,12 +39,6 @@ jobs:
         with:
           path: '.dotfiles'
           fetch-depth: 40
-      - name: Unlink git
-        shell: bash
-        if: ${{ matrix.os == 'macos-11' }}
-        run: |
-          # Make the brew link stage of git install succeed
-          brew unlink git@2.35.1 || true
       - run: |
           cd .dotfiles/
           git config --global user.email "git-perf@example.com"

--- a/git/install.sh
+++ b/git/install.sh
@@ -3,7 +3,8 @@ set -e
 
 if [ "$(uname)" == "Darwin" ]; then
   source "$DOTS/common/brew.sh"
-  brew_install git
+  # Use --overwrite due to #347, linking of git on GH Actions
+  brew_install --overwrite git
 elif [[ "$(lsb_release -i)" == *"Ubuntu"* ]]; then
   source "$DOTS/common/apt.sh"
   apt_install git

